src/app/Http/Controllers/BusinessTripController.php

```diff
@@ -77,4 +85,203 @@ public function confirm() {
     public function close() {
 
     }
+
+    public function exportPdf(int $tripId, DocumentType $documentType): JsonResponse | BinaryFileResponse
+    {
+        $trip = BusinessTrip::find($tripId);
+        if (!$trip && $tripId !== 0) {
+            return response()->json(['error' => 'Business trip not found'], 404);
+        }
+
+        $templateName = $documentType->value;
+        $templatePath = Storage::disk('pdf-templates')
+            ->path($templateName);
+
+        if (Storage::disk('pdf-templates')->missing($templateName)) {
+            Log::error("PDF template file does not exist at path: " . $templatePath);
+            return response()->json(['error' => 'PDF template file not found'], 404);
+        }
+
+        $outputPath = Storage::disk('pdf-exports')
+            ->path(uniqid('', true) . '.pdf');
+
+        $data = [];
+        switch ($documentType) {
+            case 'cestne_vyhlasenie_k_zahranicnej_pc.pdf':
+                $data = [
+                    'order_number' => $trip->sofia_id,
+                    'trip_duration' => '?', //$trip->'?', // tu potrebujem pomôcť, ktorý stĺpec z databázy to je
+                    'adress' => $trip->place,
+                    'name' => $trip->user->first_name . ' ' . $trip->user->last_name,
+                ];
+                break;
+
+            case DocumentType::COMPENSATION_AGREEMENT:
+                $data = [
+                    'first_name' => $trip->user->first_name,
+                    'last_name' => $trip->user->last_name,
+                    'academic_degree' => $trip->user->academic_degrees,
+                    'address' => $trip->user->address,
+                    'contribution1' => '?', // v databáze neexistuje prepojenie na tabuľku s prínosmi pre fakultu
```

Comment:
Existuje, trosku ale komplikovanejsie, kvoli vazobnej tabulke. Robil som na to metodu `$trip->contributions`, len ten vysledok treba sparsovat, aby sa rozlozil do troch samostatnych poli.

Na skusanie odporucam `php artisan tinker`, ktory funguje ako konzola, cize sa v nom daju hned volat metody z modelov a spustat na tom php funkcie.